test/unit/lodash.spec.js

```diff
@@ -0,0 +1,690 @@
+                var element1 = params[0] instanceof Array;
+                var element2 = params[1] instanceof Function;
+
+                expect(element1).to.eql(true);
+                expect(element2).to.eql(true);
+                expect(params.length).to.eql(2);
+            });
+            it('should return false if array contains not only number', function () {
+                spy('as');
+                expect(spy).to.have.returned(false);
+                expect(spy).to.have.been.callCount(1);
+            });
+            it('should pass mock test', function () {
+                var array = params[0];
+                var testArray = [10, 12, 35, 40, 11, 16];
+                var result = _.map.apply(_, [testArray, spy]);
+                expect(array).to.have.length(6);
+                expect(array).to.not.include.members([10, 12, 35, 40, 11, 16]);
+                expect(spy).to.have.been.callCount(7);
+                expect(result).to.eql([100, 144, 35, 1600, 11, 256]);
+            });
+            it('should square only even numbers in array', function () {
+                expect(_.map.apply(_, datasets.map2())).to.eql([1, 4, 3, 16, 5, 36]);
+            });
+        });
+    });
+
+    describe('partition', function () {
+        var users = [
+            {name: 'Andrinio', age: 36, active: false},
+            {name: 'Harry', age: 40, active: true},
+            {name: 'Kodi', age: 1, active: false},
+            {name: 'Francheska', age: 61, active: true}
+        ];
+
+        describe('partition function', function () {
+            var params = datasets.partition(users);
+            var spy = sinon.spy(params[1]);
+
+            it('should match types of passing elements', function () {
+                var element = params[1] instanceof Function;
+
+                expect(element).to.eql(true);
+                expect(params.length).to.eql(2);
+            });
+            it('should pass mock test', function () {
+                var testObj = [
+                    {qq: '123', ee: 10, active: false},
+                    {qq: '456', ee: 40, active: true}
+                ];
+                var checkFunc = _.partition.apply(_, [testObj, spy]);
+
+                expect(spy).to.have.been.callCount(2);
+                expect(checkFunc).to.eql([
+                    [{qq: '456', ee: 40, active: true}],
+                    [{qq: '123', ee: 10, active: false}]
+                ]);
+            });
+            it('should split persons to two tables using active property', function () {
+                expect(_.partition.apply(_, datasets.partition(users))).to.eql([
+                    [{name: 'Harry', age: 40, active: true}, {name: 'Francheska', age: 61, active: true}],
+                    [{name: 'Andrinio', age: 36, active: false}, {name: 'Kodi', age: 1, active: false}]
+                ]);
+            });
+        });
+        describe('partition2 function', function () {
```

Comment:
Quite poor testCase, you should instead modify it to for example split 50:50 depending if specific property is set